packages/node_modules/@node-red/runtime/lib/api/flows.js

```diff
@@ -73,6 +73,10 @@ var api = module.exports = {
             if (deploymentType === 'reload') {
                 apiPromise = runtime.flows.loadFlows(true);
             } else {
+                //ensure the runtime running/stopped state matches the deploying editor. If not, then copy the _rev number to flows.rev
+                if(flows.hasOwnProperty('_rev') && !flows.hasOwnProperty('rev') && (flows.runtimeState !== "stopped" || runtime.flows.started)) {
+                    flows.rev = flows._rev
```

Comment:
Because when deploying in stopped state, rev is omitted to indicate no merge check required. However, if state is not in sync, then we will need rev (so it is provided in _rev)

I think (it's out of my head right now)

What I do remember is while testing deploys while stopped, node-red was popping toasts about merging. And worse when deploying across multiple browsers / changing state. 

If while you are looking into the revision/consolidation of runtime-state coms, please have a try with & without this part of code. 

If not, I will revise it tomorrow and try to better explain.

Reply:
> What I do remember is while testing deploys while stopped, node-red was popping toasts about merging. And worse when deploying across multiple browsers / changing state.

If you have multiple editors open and one does a deploy, regardless of the started/stopped state, then all other editors should get the notification warning asking them to merge. That's how it's meant to work.

Reply:
Yeah but it was not working as expected. I forget the exact steps and detail but it was problematic. Something like asking to merge on the same editor that had just deployed (so we definitely matching)

Reply:
Yup - I can see it happening having stripped back some of the logic.

The POST to /flows to deploy new flows is expected to return the new `rev` value of the just-deployed flows.

However, if the flows were stopped at the point in time the call was made, then the call to `flows/index.js setFlows` was not returning the new flow rev id - so the response sent back to the editor had a blank `rev`. This meant the *next* time the editor did a deploy, it would have an out of date `rev` and get the conflict warning.

This fix is to ensure the new `rev` is properly returned from that function.

That's just an FYI - I've got a refactoring of this PR underway that includes the fix.

Reply:
Ah. That make sense.